Error in Demographics module

I have just downloaded synthea and tried to follow the instructions. 
I am able to run it successfully with or without parameters for only Massachusetts. 
Any time I try to specify any other state or state [city], I get either unknown city error or unable to load demographics module. I checked the demographics.csv file and verified that the state or state city pairs I specify are actually available. Even the same example from the readme, ./run_synthea Utah "Salt Lake City" causes the same error. Can anyone help me to understand what  I am missing as i dont see this error on the issues previously

./run_synthea Utah "Salt Lake City"
> Task :run FAILED
ERROR: unable to load demographics
java.lang.Exception: The city Salt Lake City was not found in the demographics file.
        at org.mitre.synthea.world.geography.Location.<init>(Location.java:47)
        at org.mitre.synthea.engine.Generator.init(Generator.java:133)
        at org.mitre.synthea.engine.Generator.<init>(Generator.java:106)
        at App.main(App.java:83)
Exception in thread "main" java.lang.ExceptionInInitializerError
        at org.mitre.synthea.world.geography.Location.<init>(Location.java:63)
        at org.mitre.synthea.engine.Generator.init(Generator.java:133)
        at org.mitre.synthea.engine.Generator.<init>(Generator.java:106)
        at App.main(App.java:83)
Caused by: java.lang.Exception: The city Salt Lake City was not found in the demographics file.
        at org.mitre.synthea.world.geography.Location.<init>(Location.java:47)
        ... 3 more

The error is being caused by the options processing. It is checking against an empty state at: https://github.com/synthetichealth/synthea/blob/master/src/main/java/App.java#L68
but the state is initially set to Massachusetts at:
https://github.com/synthetichealth/synthea/blob/master/src/main/java/org/mitre/synthea/engine/Generator.java#L65

If you follow the option parsing logic, the initial set to DEFAULT_STATE causes it to ignore what was passed in. This means you can only ever do cities in the DEFAULT_STATE, Massachusetts.

--- a/src/main/java/org/mitre/synthea/engine/Generator.java
+++ b/src/main/java/org/mitre/synthea/engine/Generator.java
@@ -62,7 +62,7 @@ public class Generator {
     /** Maximum age of people to be generated. Defaults to 140. */
     public int maxAge = 140;
     public String city;
-    public String state = DEFAULT_STATE;
+    public String state;
   }
   
   /**
@@ -125,6 +125,10 @@ public class Generator {
                 + "' . Valid values are file, in-memory, or none.");
     }
 
+    if (o.state == null) {
+      o.state = DEFAULT_STATE;
+    }
+    
     this.options = o;
     this.random = new Random(o.seed);
     this.timestep = Long.parseLong(Config.get("generate.timestep"));

--- a/src/test/java/AppTest.java
+++ b/src/test/java/AppTest.java
@@ -5,6 +5,7 @@ import java.util.regex.Pattern;
 import org.junit.Assert;
 import org.junit.Test;
 import org.mitre.synthea.TestHelper;
+import org.mitre.synthea.engine.Generator;
 
 public class AppTest {
 
@@ -24,6 +25,7 @@ public class AppTest {
     Assert.assertTrue(output.contains("Location:"));
     Assert.assertTrue(output.contains("alive=3"));
     Assert.assertTrue(output.contains("dead="));
+    Assert.assertTrue(output.contains("Location: Bedford, Massachusetts"));
     System.setOut(original);
   }
 
@@ -43,6 +45,7 @@ public class AppTest {
     Assert.assertTrue(output.contains("alive=4"));
     Assert.assertTrue(output.contains("dead="));
     Assert.assertFalse(output.contains("y/o F"));
+    Assert.assertTrue(output.contains("Location: " + Generator.DEFAULT_STATE));
     System.setOut(original);
   }
 
@@ -59,6 +62,7 @@ public class AppTest {
     Assert.assertTrue(output.contains("Running with options:"));
     Assert.assertTrue(output.contains("Seed:"));
     Assert.assertTrue(output.contains("alive=3"));
+    Assert.assertTrue(output.contains("Location: " + Generator.DEFAULT_STATE));
     String regex = "(.\n)*(3[0-9] y/o)(.\n)*";
     Assert.assertTrue(Pattern.compile(regex).matcher(output).find());
     regex = "(.\n)*(\\(([0-9]|[0-2][0-9]|[4-9][0-9]) y/o)(.\n)*";
@@ -66,6 +70,24 @@ public class AppTest {
     System.setOut(original);
   }
 
+
+  @Test
+  public void testAppWithDifferentLocation() throws Exception {
+    TestHelper.exportOff();
+    String[] args = {"-s", "0", "-p", "3", "Utah", "Salt Lake City"};
+    final PrintStream original = System.out;
+    final ByteArrayOutputStream out = new ByteArrayOutputStream();
+    final PrintStream print = new PrintStream(out, true);
+    System.setOut(print);
+    App.main(args);
+    String output = out.toString();
+    Assert.assertTrue(output.contains("Running with options:"));
+    Assert.assertTrue(output.contains("Seed:"));
+    Assert.assertTrue(output.contains("alive=3"));
+    Assert.assertTrue(output.contains("Location: Salt Lake City, Utah"));
+    System.setOut(original);
+  }
+  
   @Test
   public void testInvalidArgs() throws Exception {
     String[] args = {"-s", "foo", "-p", "foo", "Massachusetts", "Bedford"};

